sw/device/lib/dif/dif_pwrmgr.h

```diff
@@ -0,0 +1,499 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_LIB_DIF_DIF_PWRMGR_H_
+#define OPENTITAN_SW_DEVICE_LIB_DIF_DIF_PWRMGR_H_
+
+/**
+ * @file
+ * @brief <a href="/hw/ip/pwrmgr/doc/">Power Manager</a> Device Interface
+ * Functions
+ */
+
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_warn_unused_result.h"
+#include <stddef.h>
+#include <stdint.h>
+
+// Header Extern Guard (so header can be used from C and C++)
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+/**
+ * Internal state of a power manager.
+ *
+ * Instances of this struct must be initialized by `dif_pwrmgr_init()` before
+ * being passed to other functions in this library. Its fields should be
+ * considered private and are only provided here so that callers can allocate
+ * it.
+ */
+typedef struct dif_pwrmgr {
+  mmio_region_t base_addr;
+  uint8_t num_wakeup_sources;
+  uint8_t num_reset_sources;
```

Comment:
Consider just keeping around the `config` struct, so that you don't need to redefine this.